src/native/_system_uri.js

```diff
@@ -11,24 +12,38 @@ let isSysUriLibLoadErr = null;
 
 const init = (options) => {
   ffi = FFI.Library(path.join(options.libPath || dir, SYSTEM_URI_LIB_FILENAME), {
-    open: [t.i32, ['string'] ],
-    install: [t.i32, ['string', //bundle
+    open: ["void", ['string', 'pointer', 'pointer'] ],
+    install: ["void", ['string', //bundle
       'string', //vendor
       'string', //name
       'string', //exec
       'string', //icon
-      'string', //schemes
+      'string', //schemes,
+      'pointer', // userdata
+      'pointer'
     ] ],
   });
 };
 
+function _handleError() {
+  return FFI.Callback("void", [t.VoidPtr, t.FfiResult],
```

Comment:
This will throw error out of the call stack. instead return promises as we do for all other APIs